### DISCUSSION TITLE
ST support

### DISCUSSION THREAD

**Discussion by mahexyz** [hathach/tinyusb#2005] [General]
I want to use the stack on an stm32h7 as host. But currently I have some trouble or missunderstandings.
Can somebody please give me advice?

1) In the current master branch, I can only find an empty .gitmodules file. Dependencies are not downloaded.
2) the examples for stm32 host are not building, when using the make command for the nucleo board. Some linkage errors occur.
3) as far as I understand, I need the ohci.c file. but this is dependend on a base address of a lpc device. The exchange to an ST address is not possible for my opinion.
4) the dcd_synopsys.c file is needed. But this file implements only device functionalities and not host functionalities, as far as I understand.
5) Is there a tutorial or an example for my purposes which I have not seen? Was the stack ever tested or used for stm32 hosts?

Thanks a lot for your help and best regards

**Comment by kf6gpe** [comment]
Howdy!

We just got the audio code running in minimal sample created by Cube about an hour ago.

For your specific questions.
1. You should be able to `git submodule` to get the submodules you want. I don't remember how exactly I did it. Something like
```
  make BOARD=stm32h745disco get-deps 
  make BOARD=stm32h745disco all
```
2. I have a disco board and got the audio test example building pretty quickly. I followed the baseline instructions, if I recall correctly.
3. I did not use `ochi.c` at all in either the audio sample or the port to our h/w.
4. Yes, only device is supported for the STM37H7.
5. I got things running for a minimal audio test, but unfortunately the code's wrapped up in some of our other stuff and I can't share it. I can confirm that starting with CubeMX and then folding what you need may be an option; that's what got us going.

Hope that helps.

**Comment by hathach** [comment]
> In the current master branch, I can only find an empty .gitmodules file. Dependencies are not downloaded.

submodules is removed and dependencies are now managed by an script https://github.com/hathach/tinyusb/blob/master/tools/get_deps.py . Since tinyusb is often included as submodules to other projects e.g pico-sdk and people is often init submodule with recursive and complain that it takes too long due to lots of mcu submodule.

> Is there a tutorial or an example for my purposes which I have not seen? Was the stack ever tested or used for stm32 hosts?

stm32 host is not supported yet.
